src/udiskslinuxpartition.c

```diff
@@ -476,8 +428,12 @@ handle_set_name (UDisksPartition       *partition,
                                                     invocation))
     goto out;
 
-  device_name = udisks_daemon_util_escape_and_quote (udisks_block_get_device (partition_table_block));
-  escaped_name = udisks_daemon_util_escape_and_quote (name);
+  device_name = g_strdup (udisks_block_get_device (partition_table_block));
+  partition_name = g_strdup (udisks_block_get_device (block));
+
+  /* hold a file descriptor open to suppress BLKRRPART generated by the tools */
+  fd = open (udisks_block_get_device (block), O_RDONLY);
```

Comment:
Storaged holds the flock only when creating the partition, this code is for editting name/type of an existing partition.